### PR TITLE
Full Sweep: Drop the USE_UPDATED_SIGNING_UI feature flag

### LINKED ISSUE #3449: Feature flag cleanup
A bunch of feature flags have been enabled for some time but still exist and have alternate code paths. We want to clean these up so we can make new ones a little easier to manage. 

Notable flags are the NFT set and the new signing UI flag.

### PR DESCRIPTION
**Pull request by Shadowfiend**
This takes care of the remaining cleanup from the old
signing flow:
- A couple of now-unused files.
- Some type refinements and specificity improvements.
- Dropping keyring-path specific stuff in Popup.tsx.

And of course the removal of the feature flag itself.

Along the way I also fixed a couple of components that
were using `React.FC` and `export const` in a way that
we don't do anywhere else.

## Testing

- [x] Make sure transaction signing continues to work
      correctly for Ledger and non-Ledger flows :)

Closes #3449 .

Latest build: [extension-builds-3475](https://github.com/tahowallet/extension/suites/13699362199/artifacts/757434807) (as of Mon, 19 Jun 2023 09:38:28 GMT).

### CODE REVIEW

**Review by jagodarybacka — COMMENTED** [review]
Code is fine, will proceed to test it manually

File: ui/components/SignData/EIP4361Info.tsx [hunk 1]
```diff
@@ -1,36 +1,22 @@
...
-const EIP4361Info: React.FC<{
-  signingData: EIP4361Data
-  // FIXME Drop this once new signing flow is final.
-  excludeHeader?: boolean
-}> = ({ signingData, excludeHeader = false }) => {
+export default function EIP4361Info({ signingData }: Props): ReactElement {
   const { t } = useTranslation("translation", { keyPrefix: "signing.EIP4361" })
+
+  const chainName =
+    NETWORK_BY_CHAIN_ID[signingData.chainId.toString()]?.name ?? "Unknown"
```
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply ui/components/SignData/EIP4361Info.tsx:16]: Yeah agreed… Could even use `signTransaction.unknownNetwork` tbh, but the PR already started to get away from me a little 😬

File: ui/components/SignData/EIP191Info.tsx [hunk 2]
```diff
@@ -1,27 +1,20 @@
...
   const { t } = useTranslation("translation", { keyPrefix: "signing" })
+
   return (
     <>
-      {excludeHeader ? (
-        <></>
-      ) : (
-        <>
-          <div className="label header">
-            {internal ? t("signatureRequired") : t("dappSignatureRequest")}
```
Comment (kkosiorowska) [inline ui/components/SignData/EIP191Info.tsx:20]: We can remove `dappSignatureRequest` form messages.json. It is no longer used.

Comment (Shadowfiend) [reply ui/components/SignData/EIP191Info.tsx:20]: https://github.com/tahowallet/extension/pull/3475/commits/7d4eb347deca98173ee1534ae0eccbce66372f92

**Review by jagodarybacka — APPROVED** [review]
All good, manually tested:
- signing messages with and without Ledger
- send with and without Ledger
- swap with and without Ledger
🚀

**Pull request merged**